app/logic/Mail/SQL/SQLMailAccount.ts

```diff
@@ -85,13 +79,10 @@ export class SQLMailAccount {
         backgroundError(ex);
       }
     }
-    // Set SMTP accounts
-    for (let account of accounts) {
-      let id = account.outgoingAccountID;
-      delete account.outgoingAccountID;
-      if (id) {
-        account.outgoing = smtpAccounts.find(acc => acc.id == id);
-      }
+    // Need to special-case it here, because SMTP accounts
+    // aren't in `appGlobal.emailAccounts`, so `setMainAccounts()` won't find them.
+    for (let smtp of smtpAccounts) {
+      smtp.mainAccount = accounts.find(acc => acc.id == smtp._mainAccountID);
```

Comment:
(I was wondering how this worked with other cases, but I think what happens is that this causes the account to be the outgoing account of its main account, at which point it will start appearing in `allAccounts()`.)